### ISSUE TITLE
document high-level usage of dub as a library

### ISSUE THREAD

**Issue by timotheecour** [dlang/dub#1017]
Could we have (even a starting point of) example usage of dub as a library (say in README)?
Something a little more helpful than [https://github.com/dlang/dub#using-dub-as-a-library] would be great; eg:

* how would I generate a $PackageRecipe from a package name? etc

**Comment by LaurentTreguier** [comment]
Agreed. The struct `Path` from vibe.d can be used instead by using the `library-nonet` configuration, which means `import dub.internal.vibecompat.inet.path` can be replaced with `import vibe.core.path`; but right now dub doesn't compile with vibe.d if the `vibe-core` subConfiguration is selected, so this solution is not yet perfect.

**Comment by s-ludwig** [comment]
For reference, there is an open PR for vibe-core compatibility: #1283
